account_class.rb

```diff
@@ -0,0 +1,58 @@
+module Bank
+  class Account
+    attr_reader :account_id, :balance, :time
+    #allows account to be created with an initial balance as a parameter
+    def initialize(balance)
+      #creates ID#
+      @account_id = generate_id
+      @balance = balance_restriction(balance)
+      @time = account_opened
+
+
+      #prevents initial balance from being a negative amount
+      #if balance < 0
+      #  raise Exception.new, "No negative balance allowed! Live within your means!"
+      #end
+    end
+
+  #method to create an ID number, not unique number yet
+
+    def balance_restriction (balance)
+      if balance < 0
+        raise Exception.new("No negative balance allowed! Live within your means!")
+      else
+        return balance
+      end
+    end
+    
+    def generate_id
+      user_id = rand(1000..9000)
+      return user_id
+    end
+
+    def account_opened
+      time = Time.new
+      return time
+    end
+
+    #method for withdrawal
+    def withdrawal(subtract_money)
+      if (@balance - subtract_money) >= 0
+        @balance = (@balance - subtract_money)
```

Comment:
Since you are using this logic twice, I'd recommend using a variable to store this result so it does not to be calculated twice